src/ComposerManager.php

```diff
@@ -3,37 +3,48 @@
 
 namespace LotGD\Core;
 
-use Composer\Composer;
+use Composer\{
+    Composer,
+    Factory,
+    IO\NullIO
+};
 use Monolog\Logger;
 
-use LotGD\Core\Exceptions\LibraryDoesNotExistException;
+use LotGD\Core\{  
+    Exceptions\InvalidConfigurationException,
+    Exceptions\LibraryDoesNotExistException
+};
 
 /**
  * Helps perform tasks with the composer configuration.
  */
 class ComposerManager
 {
-    private $logger;
     private $composer;
 
     /**
-     * Creates a new ComposerManager.
-     * @param Monlog\Logger $logger A logger instance for messaging.
-     */
-    public function __construct(Logger $logger)
-    {
-        $this->logger = $logger;
-    }
-
-    /**
      * Returns a Composer instance to perform underlying operations on. Be careful.
      * @return Composer An instance of Composer.
      */
     public function getComposer(): Composer
     {
         if ($this->composer === null) {
-            $this->composer = \Composer\Factory::create(new \Composer\IO\NullIO());
+            // Search "true" working directory
```

Comment:
Symfony's working directory is not the project root, but /web (since the entry point to the application is web/app.php). The web directory lacks a composer.json file, leading to an exception here.

This approach is a bit more elegant than just checking if it ends with /web.